src/pkgcheck/sources.py

```diff
@@ -59,6 +60,19 @@ def itermatch(self, restrict, sorter=sorted, **kwargs):
         return self.source.itermatch(restrict, sorter=sorter, **kwargs)
 
 
+class LatestVersionRepoSource(RepoSource):
+    """Repo source that returns only the latest non-VCS and VCS slots"""
+
+    def itermatch(self, *args, **kwargs):
+        for _, pkgs in groupby(super().itermatch(*args, **kwargs),
+                               key=lambda pkg: pkg.slotted_atom):
+            best_by_live = {}
+            for pkg in pkgs:
+                best_by_live[pkg.live] = pkg
+            for pkg in sorted(best_by_live.values()):
+                yield pkg
```

Comment:
For better context in case an error occurs:
`yield from sorted(best_by_live.values())`